nomad/structs/structs.go

```diff
@@ -4672,6 +4672,13 @@ func (j *Job) Validate() error {
 				fmt.Errorf("Job task group %s has count %d. Count cannot exceed 1 with system scheduler",
 					tg.Name, tg.Count))
 		}
+
+		if tg.MaxClientDisconnect != nil &&
+			tg.ReschedulePolicy.Attempts > 0 &&
+			tg.PreventRescheduleOnLost {
+			err := fmt.Errorf("max_client_disconnect and prevent_reschedule_on_lost cannot be enabled when rechedule.attempts > 0")
+			mErr.Errors = append(mErr.Errors, err)
+		}
```

Comment:
I'm still not sure about this validation as they seem to mix two scenarios. For example, a group configured like this:

```hcl
group "..." {
  max_client_disconnect      = "12h"
  prevent_reschedule_on_lost = true
  
  reschedule {
    attempts = 10
  }
}
```

As an user, I think my intention in writing this job would be something like this:

- Try to reschedule my allocations at least 10 times if they fail.
- If a node running an alloc for this group misses heartbeats, consider it `disconnected` instead of `down` for 12h as it may come back up, but don't create a replacement allocation. After the 12h consider the allocation as `lost` and create a replacement.

So in a scenario where this group has 3 allocs, if a node misses its hearbeats I would expect there to still be 3 allocs, 2 `running` and 1 `unknown`. After 12h, if the node doesn't reconnect, it will considered `down` and the alloc `lost`, and a replacement is created, resulting in 4 allocs: 3 `running` and 1 `lost`.

This is consistent with the documented behaviour of the `reschedule` block:
> Nomad will attempt to schedule the allocation on another node if any of its task statuses become `failed`.

Since the alloc is `unknown`, none of the tasks should be in `failed` state, so the `reschedule` block doesn't apply. By forcing `reschedule.attempts` to be 0 we're preventing users from being able to handle allocation failures.

In hindsight, I think the confusion started with `max_client_disconnect`, where the docs mention:
> Replacement allocations will be scheduled according to the allocations' reschedule policy until the disconnected client reconnects. 

This conflates cluster state (a node misses heartbeat) with allocation state (a task fails), making it impossible to properly handle both cases as any combination of them is valid.

If we were to refactor this part of the code I think I would create a new block to disentangle these two scenarios. `reschedule` is applied when allocations fail, and a new `disconnected` (or a better name 😅) block to handle nodes missing heartbeats.

This new block could then have configurations like these:

* Create a replacement for `unknown` allocs but try to reconnect the original alloc if the node comes back within 12h (equivalent to today's `max_client_disconnect`).
    ```hcl
    disconnected {
      replace_unknown     = true
      max_disconnect_time = "12h"
    }
    ```

* Never create a replacement for `unkown` allocs (I think this is what we're trying to accomplish here with `prevent_reschedule_on_lost`).
    ```hcl
    disconnected {
      replace_unknown = false
    }
    ```

* Wait 12h before creating a replacement for `unknown` allocs (I don't think this is currently possible to express, even with  this PR, and I think it's a valid use case).
    ```hcl
    disconnected {
      replace_unknown     = false
      max_disconnect_time = "12h"
    }
    ```

Parallel to all of these configs, I could set different `reschedule` policies to handle allocations that fail. Regardless of client status, I probably do want to reschedule them (or maybe I don't 😄).

This is a _lot_ to change that would require proper planning and scoping, but until then I think the relationship between all of these flags can get confusing.

For this PR specifically, if the intention is to prevent the use of `max_client_disconnect` and `prevent_reschedule_on_lost` with a `reschedule` policy that allows for new allocations to be created, then I think we also need to check the value of `reschedule.unlimited`, as its often used with with `reschedule.attempts = 0` but does create replacements.